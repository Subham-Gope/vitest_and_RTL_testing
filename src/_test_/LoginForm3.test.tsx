// import { describe, test, expect } from "vitest";
// import { render, screen } from "@testing-library/react";
// import userEvent from "@testing-library/user-event";
// import LoginForm3 from "../components/LoginForm3";

// describe("Testing the LoginForm3.tsx file", () => {
//   test("move from one interactable element to the other using the tab", async () => {
//     // setting up the userEvent
//     const user = userEvent.setup();

//     // Setting up the debugger
//     screen.debug();

//     // setting up the snapshot matching
//     const { container } = render(<LoginForm3 />);
//     expect(container).toMatchSnapshot();

//     const username = screen.getByPlaceholderText(/username/i);
//     const password = screen.getByPlaceholderText(/password/i);
//     const btn = screen.getByRole("button", { name: /submit/i });

//     expect(username).toBeInTheDocument();
//     expect(password).toBeInTheDocument();
//     expect(btn).toBeInTheDocument();

//     // Initiating the user interactions
//     await user.tab();
//     expect(username).toHaveFocus();
//     await user.tab();
//     expect(password).toHaveFocus();
//     await user.tab();
//     expect(btn).toHaveFocus();

//   });
// });
