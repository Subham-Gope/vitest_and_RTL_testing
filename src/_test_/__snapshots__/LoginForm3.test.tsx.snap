// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing the LoginForm3.tsx file > move from one interactable element to the other using the tab 1`] = `
<div>
  <form>
    <input
      placeholder="username"
      type="text"
    />
    <input
      placeholder="password"
      type="text"
    />
    <button>
      Submit
    </button>
  </form>
</div>
`;
